### PR TITLE
warn before abandoning unsaved changes

### PR DESCRIPTION
**Pull request by zackgalbreath**
Confirm with the user when they're about to navigate away from unsaved changes to an analysis.

### CODE REVIEW

File: app/src/views/AnalysisManagementView.js [hunk 1]
```diff
@@ -181,6 +182,24 @@
...
+                    $("#save").removeClass("disabled");
+                    if (window.onbeforeunload === null) {
+                        window.onbeforeunload = function (e) {
+                            return "You have unsaved changes in the editor.";
+                        };
+                    }
+                } else {
+                    window.onbeforeunload = null;
+                    $("#save").removeClass("btn-primary");
+                    $("#save").addClass("disabled");
```
Comment (zachmullen) [inline app/src/views/AnalysisManagementView.js]: Not a big deal, but just for future reference, jquery provides nice chainable semantics to reduce the need to run the same selector multiple times in a row, so the above two lines could be combined into

    $('#save').removeClass("btn-primary").addClass("disabled");

**Comment by zachmullen** [comment]
LGTM

**Comment by zackgalbreath** [comment]
Any idea on how to fix the failing integration checks?  Maybe it's just me, but I experience the same failure locally on master...

**Comment by zachmullen** [comment]
I haven't looked at the tests in this repo at all, but I've found the best way to debug phantomjs tests is to have it generate screenshots. In Girder, we have [some code](https://github.com/girder/girder/blob/master/clients/web/test/specRunner.js#L127-L142) in our phantom execution wrapper that writes a screenshot to disk if there is a catastrophic failure. We also [listen to console output](https://github.com/girder/girder/blob/master/clients/web/test/specRunner.js#L58-L67) so test writers can add `console.log('__SCREENSHOT__');` to easily inspect the app state at any point during test execution. We make use of that feature [in our actual runtime code](https://github.com/girder/girder/blob/master/clients/web/test/lib/jasmine-1.3.1/ConsoleReporter.js#L98-L101) by hooking into jasmine's reporting system and generating a screenshot if any test spec fails and generating a timestamped file that gets sent to CDash via magic output.

I created that system before I realized there were better ways to communicate with the outside phantom runner besides console output. I'd recommend using [onCallback](https://github.com/girder/girder/blob/master/clients/web/test/specRunner.js#L84) to signal the runner instead.

**Pull request merged**